langserver/handlers/symbols.go

```diff
@@ -39,22 +38,10 @@ func (h *logHandler) TextDocumentSymbol(ctx context.Context, params lsp.Document
 
 	// TODO: block until it's available <-pf.ParserLoadingDone()
 	// requires https://github.com/hashicorp/terraform-ls/issues/8
-	// TODO: replace crude wait/timeout loop
-	var isParserLoaded bool
-	var elapsed time.Duration
-	sleepFor := 10 * time.Millisecond
-	maxWait := 3 * time.Second
-	for !isParserLoaded {
-		time.Sleep(sleepFor)
-		elapsed += sleepFor
-		if elapsed >= maxWait {
-			return symbols, fmt.Errorf("parser is not available yet for %s", file.Dir())
-		}
-		var err error
-		isParserLoaded, err = pf.IsParserLoaded(file.Dir())
-		if err != nil {
-			return symbols, err
-		}
+	if err := Waiter(func() (bool, error) {
```

Comment:
Nitpick: Do you mind just mentioning here in a comment why do we need to wait? I assume because the request for symbols comes early after `didOpen` and a valid response is expected (i.e. there is no way to respond "try again later").

Reply:
Will do